pkg/deploy/api/types.go

```diff
@@ -121,6 +121,9 @@ type ExecNewPodHook struct {
 	// Volumes is a list of named volumes from the pod template which should be
 	// copied to the hook pod.
 	Volumes []string
+	// MaxRestarts specifies a number of allowed lifecycle pod container restarts
+	// before we treat the lifecycle hook as failed.
+	MaxRestarts int
```

Comment:
i thought it should be close to the pod definition... but I agree that having it on LifecycleHook might better for validation.